examples/decoding/plot_ssd_spatial_filters.py

```diff
@@ -0,0 +1,442 @@
+        """
+        self._check_X(X)
+        if self.filters_ is None:
+            raise RuntimeError('No filters available. Please first call fit')
+        if X.ndim == 2:
+            X_ssd = np.dot(self.filters_.T, X[self.picks_])
+        elif X.ndim == 3:
+            # project data on source space
+            X_ssd = np.empty_like(X)
+            for ii, x in enumerate(X[:, self.picks_]):
+                X_ssd[ii] = np.dot(self.filters_.T, x)
+        else:
+            raise NotImplementedError()
+
+        if self.sort_by_spectral_ratio:
+            self.spec_ratio, self.sorter_spec = self.spectral_ratio_ssd(
+                ssd_sources=X_ssd)
+            self.filters_ = self.filters_[:, self.sorter_spec]
+            self.patterns_ = self.patterns_[self.sorter_spec]
+
+            if X.ndim == 2:
+                X_ssd = X_ssd[self.sorter_spec][:self.n_components]
+            else:
+                X_ssd = X_ssd[:, self.sorter_spec, :][:, :self.n_components, :]
+        return X_ssd
+
+    def spectral_ratio_ssd(self, ssd_sources):
+        """
+        Spectral ratio measure for best n_components selection
+        See :footcite:`NikulinEtAl2011`, Eq. (24).
+
+        Parameters
+        ----------
+        ssd_sources : array
+            data proyected on source space.
+
+        Returns
+        -------
+        spec_ratio : array, shape (n_channels)
+            array with the sprectal ratio value for each component
+        sorter_spec : array, shape (n_channels)
+            array of indeces for sorting spec_ratio.
+
+        """
+        psd, freqs = psd_array_welch(
+            ssd_sources, sfreq=self.info['sfreq'], n_fft=self.n_fft)
+        sig_idx = _time_mask(freqs, *self.freqs_signal)
+        noise_idx = _time_mask(freqs, *self.freqs_noise)
+        if psd.ndim == 3:
+            mean_sig = psd[:, :, sig_idx].mean(axis=2).mean(axis=0)
+            mean_noise = psd[:, :, noise_idx].mean(axis=2).mean(axis=0)
+            spec_ratio = mean_sig / mean_noise
+        else:
+            mean_sig = psd[:, sig_idx].mean(axis=1)
+            mean_noise = psd[:, noise_idx].mean(axis=1)
+            spec_ratio = mean_sig / mean_noise
+        sorter_spec = spec_ratio.argsort()[::-1]
+        return spec_ratio, sorter_spec
+
+    def apply(self, inst):
+        """Remove selected components from the signal.
+
+        This procedure will reconstruct M/EEG signals from which the dynamics
+        described by the excluded components is subtracted
+        (denoised by low-rank factorization).
+        See :footcite:`HaufeEtAl2014` for more information.
+
+        The data is processed in place.
+
+        Parameters
+        ----------
+        inst : instance of Raw, Epochs or ndarray
+             The data to be processed. The instance is modified inplace.
+
+        Returns
+        -------
+        X : instance of Raw, Epochs or ndarray
+            The processed data.
+        """
+        X_ssd = self.transform(inst)
```

Comment:
We should probably sort by spectral ratio -- on the fly -- here, if demanded by option. Agreed?

Reply:
perfect!